### ISSUE TITLE
Problem in Celery with SQLA

### ISSUE THREAD

**Issue by giovannipizzi** [aiidateam/aiida_core#464]
The daemon crashes at the first connection with SQLA.
Some random comments:
- We cannot upgrade to celery 4 because SQLA is not valid anymore as a broker
- We are using SQLA as a broker currently
- It is possible that there is a conflict between the SQLA connections of Celery and ours?
- We should go away from SQLA as a broker. Two ways: 1. we upgrade, and use RabbitMQ (note that however this requires a new service running on users' computers); 2. We stop using Celery - in the end, we don't really need it, we just need to run four functions every few seconds.
- Upgrading to the most recent 3.x version of celery does not solve the issue, see giovannipizzi/aiida_core@454db10c6d

**Comment by giovannipizzi** [comment]
Celery docs say that basically we shouldn't use the SQLA broker... http://docs.celeryproject.org/en/3.1/getting-started/brokers/sqlalchemy.html
Also interesting to note that 4.x drops support

**Comment by giovannipizzi** [comment]
Should we just replace with the python standard library `sched`? https://docs.python.org/2/library/sched.html

**Comment by szoupanos** [comment]
Do we have an example which shows that it crashes during the first connection with SQLA as a broker?
(Because my experience is the opposite.)
Maybe things became worse after we activated the schema check in one of the latest commits.
I will try to add references to this ticket too.

**Comment by szoupanos** [comment]
Moreover, I also agree with the comment that we may also stop using Celery (if it Celery's problem and not ours). We can also launch ourselves a few functions every few seconds and it might be better than depending on 3rd party software for something simple. Especially if it introduces conflicts, unpredictable behaviour or prohibits us to upgrade more important packages (like SQLA etc)

**Comment by giovannipizzi** [comment]
You're right, it's not the first connection, but it typically happens quite early (after a few commands).

I'm working in my fork, today with Fernando we've replaced the session with a scoped session, but this still didn't solve the issue.

**Comment by szoupanos** [comment]
I just checked out the develop branch (commit 0914ed9) and indeed the daemon crashes as soon as it starts.

I am a bit surprised that that I did not observe the problem when I checked the pull request #422.
I assume this happened because of what you mentioned in your email - that I was testing the changes of that pull request using a VM with one assigned CPU.

Now, I use 2 and I get immediately the exceptions.

Apart from the above, the exception that we get here is the one that was related to the DbSettigns table. As soon as the schema check is disabled, I don't get any exceptions and I manage to submit calculations successfully etc.

I will also investigate more this issue since it is time to end with this problem.
